pkg/xen-tools/initrd/udhcpc_script.sh

```diff
@@ -35,14 +35,22 @@ case "$1" in
     # configure interface and routes
     ip addr flush dev $interface
     ip addr add ${ip}/${mask} dev $interface
-    [ -n "$router" ] && ip route add default via ${router%% *} dev $interface
+    [ -n "$router" ] && ip route add ${router%% *} dev $interface && ip route add default via ${router%% *} dev $interface
     # setup dns
     if [ "$interface" == "$PEERDNS_IF" ] ; then
       [ -n "$domain" ] && echo search $domain > $RESOLV_CONF
       for i in $dns ; do
         echo nameserver $i >> $RESOLV_CONF
       done
     fi
+    if [ -n "${hostname}" ] && [ "${hostname}" != "(none)" ]; then
```

Comment:
With multiple interfaces (just pulled @rvs 's PR for that) this code will be executed for each interface. That might add duplicate entries in /etc/hosts

Reply:
If the script can check if an entry already exists in /etc/hosts before adding it we can handle that.